sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/FilterPushdownBenchmark.scala

```diff
@@ -378,5 +409,60 @@ object FilterPushdownBenchmark extends BenchmarkBase with SQLHelper {
         }
       }
     }
+
+    runBenchmark(s"Predicate conversion benchmark with unbalanced Column") {
+      // This benchmark tests a very isolated part of the predicate pushdown process - specifically,
+      // the individual action of converting a Spark `Expression` to an ORC `SearchArgument`.
+      // This results in more granular numbers that can help highlight small performance
+      // differences in this part of the code that would be hidden by slower components that
+      // get run when a full Spark job is executed.
+      // The benchmark below runs a more complete, end-to-end test which covers the whole pipeline
+      // and can uncover high-level performance problems, but is bad at discriminating details.
+      val numRows = 160000
+      val width = 2000
+
+      val columns = (1 to width).map(i => s"id c$i")
+      val df = spark.range(numRows).selectExpr(columns: _*)
+      val benchmark = new Benchmark(
+        s"Convert filters to ORC filter",
+        numRows, minNumIters = 5, output = output)
+      Seq(2500, 5000, 15000).foreach { numFilter =>
+        val whereColumn = (1 to numFilter)
+          .map(i => col("c1") === lit(i))
+          .foldLeft(lit(false))(_ || _)
+
+        benchmark.addCase(s"Convert $numFilter filters to ORC filter") { _ =>
+          OrcFilters.createFilter(df.schema,
+            DataSourceStrategy.translateFilter(whereColumn.expr).toSeq)
+        }
+      }
+      benchmark.run()
+    }
+
+    runBenchmark(s"Pushdown benchmark with unbalanced Column") {
+      val numRows = 160000
+      val width = 200
+
+      withTempPath { dir =>
+        val columns = (1 to width).map(i => s"id c$i")
+        val df = spark.range(numRows).selectExpr(columns: _*)
+        withTempTable("orcTable", "parquetTable") {
+          saveAsTable(df, dir)
+          val benchmark =
+            new Benchmark("Select data with filters", numRows, minNumIters = 5, output = output)
+          Seq(100, 500, 1000).foreach { numFilter =>
```

Comment:
I tried with 5000 filters, and the execution becomes very slow. For end-to-end tests, we need to have a smaller size here, comparing to the benchmark `Convert filters to ORC filter`